src/components/InLineNotification.tsx

```diff
@@ -0,0 +1,158 @@
+import React from 'react'
+import { StyleProp, StyleSheet, Text, TextStyle, View, ViewStyle } from 'react-native'
+import AttentionIcon from 'src/icons/Attention'
+import Colors from 'src/styles/colors'
+import fontStyles, { typeScale } from 'src/styles/fonts'
+import { Spacing } from 'src/styles/styles'
+
+export enum Severity {
+  Informational,
+  Warning,
+  Error,
+}
+
+interface Props {
+  severity: Severity
+  title?: string | null
+  description: string
+  style?: StyleProp<ViewStyle>
+  ctaLabel?: string | null
+  onPressCta?: () => void
+  ctaLabel2?: string | null
+  onPressCta2?: () => void
+  testID?: string
+}
+
+interface CustomStyles {
+  container: ViewStyle
+  attentionIcon: Record<'color', Colors>
+  ctaLabel: TextStyle
+  ctaLabel2: TextStyle
+}
+
+export function InLineNotification({
+  severity,
+  title,
+  description,
+  style,
+  ctaLabel,
+  onPressCta,
+  ctaLabel2,
+  onPressCta2,
+  testID,
+}: Props) {
+  const severityStyle = severityStyles[severity]
+  return (
+    <View style={[defaultStyles.container, severityStyle.container, style]} testID={testID}>
+      {title && (
+        <View style={defaultStyles.row}>
+          <View style={defaultStyles.attentionIcon}>
+            <AttentionIcon color={severityStyle.attentionIcon.color} />
+          </View>
+          <Text style={defaultStyles.titleText}>{title}</Text>
+        </View>
+      )}
+
+      <View style={defaultStyles.bodyContainer}>
+        <View style={[defaultStyles.row, defaultStyles.bodyRow]}>
+          <View style={defaultStyles.attentionIcon}>
+            {!title && <AttentionIcon color={severityStyle.attentionIcon.color} />}
+          </View>
+          <Text style={defaultStyles.bodyText}>{description}</Text>
+        </View>
+
+        {(ctaLabel || ctaLabel2) && (
+          <View style={[defaultStyles.row, defaultStyles.ctaRow]}>
+            {ctaLabel && onPressCta && (
+              <Text style={[defaultStyles.ctaLabel, severityStyle.ctaLabel]} onPress={onPressCta}>
+                {ctaLabel}
+              </Text>
+            )}
+            {ctaLabel2 && onPressCta2 && (
+              <Text
+                style={[defaultStyles.ctaLabel2, severityStyle.ctaLabel2]}
+                onPress={onPressCta2}
+              >
+                {ctaLabel2}
+              </Text>
+            )}
+          </View>
+        )}
+      </View>
+    </View>
+  )
+}
+
+const defaultStyles = StyleSheet.create({
+  container: {
+    paddingHorizontal: Spacing.Regular16,
+    paddingTop: Spacing.Regular16,
+    paddingBottom: Spacing.Small12,
+    borderRadius: Spacing.Regular16,
+    width: '100%',
+    gap: Spacing.Tiny4,
+  },
+  bodyContainer: {
+    gap: Spacing.Smallest8,
+  },
+  row: {
+    flexDirection: 'row',
+    gap: Spacing.Smallest8,
+    alignItems: 'center',
+  },
+  bodyRow: {
+    alignItems: 'flex-start',
+  },
+  ctaRow: {
+    paddingVertical: Spacing.Tiny4,
+    paddingHorizontal: Spacing.Smallest8,
```

Comment:
are you sure you want padding horizontal here? doesn't that mean the right hand side of the button would be indented on the right compared to the description (i.e. the content doesn't line up on the right)?

Reply:
That's correct. I think it's more visually pleasing to keep this padding, helps to have a focus on the button options. Figma design matches this as well.